cpp/src/tsne/distances.cuh

```diff
@@ -91,16 +111,17 @@ void normalize_distances(const int n, float *distances, const int n_neighbors,
  * @param[in] stream: The GPU stream.
  * @param[in] handle: The GPU handle.
  */
-void symmetrize_perplexity(float *P, long *indices, const int n, const int k,
-                           const float exaggeration,
+template <typename knn_value_idx>
+void symmetrize_perplexity(float *P, knn_value_idx *indices, const int n,
+                           const int k, const float exaggeration,
                            MLCommon::Sparse::COO<float> *COO_Matrix,
                            cudaStream_t stream, const raft::handle_t &handle) {
   // Perform (P + P.T) / P_sum * early_exaggeration
   const float div = exaggeration / (2.0f * n);
   raft::linalg::scalarMultiply(P, P, div, n * k, stream);
 
   // Symmetrize to form P + P.T
-  MLCommon::Sparse::from_knn_symmetrize_matrix(
+  MLCommon::Sparse::from_knn_symmetrize_matrix<float, 32, 32, knn_value_idx>(
```

Comment:
Can we put `knn_value_idx` first or second in the template list so we don't have to hardcode 32 here?